fails to build with c-interface feature on systems where char is unsigned.

While working on the package of your crate in Debian, I discovered it fails to build if the c-interface feature is enabled and the architecture is one where the c type char is unsigned.

https://ci.debian.net/data/autopkgtest/testing/arm64/r/rust-sysinfo/23060510/log.gz

    error[E0308]: mismatched types
       --> src/c_interface.rs:467:39
        |
    467 |             let _ = CString::from_raw(s as usize as *mut i8);
        |                                       ^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
        |
        = note: expected raw pointer `*mut u8`
                   found raw pointer `*mut i8`

The fix is trivial, just replace i8 with c_char. https://salsa.debian.org/rust-team/debcargo-conf/-/blob/master/src/sysinfo/debian/patches/fix-char-types.diff

--- a/src/c_interface.rs
+++ b/src/c_interface.rs
@@ -462,7 +462,7 @@ pub extern "C" fn sysinfo_process_current_directory(process: CProcess) -> RStrin
 pub extern "C" fn sysinfo_rstring_free(s: RString) {
     if !s.is_null() {
         unsafe {
-            let _ = CString::from_raw(s as usize as *mut i8);
+            let _ = CString::from_raw(s as usize as *mut _);
         }
     }
 }

